.github/workflows/src/verify-run-status.js

```diff
@@ -0,0 +1,181 @@
+import { extractInputs } from "./context.js";
+import { PER_PAGE_MAX } from "./github.js";
+
+/**
+ * @typedef {import('@octokit/plugin-rest-endpoint-methods').RestEndpointMethodTypes} RestEndpointMethodTypes
+ * 
+ * @typedef {RestEndpointMethodTypes["checks"]["listForRef"]["parameters"]} ListCheckRunsForRefParameters
+ * @typedef {RestEndpointMethodTypes["checks"]["listForRef"]["response"]["data"]["check_runs"][number]} CheckRun
+
+ * @typedef {RestEndpointMethodTypes["actions"]["listWorkflowRunsForRepo"]["parameters"]} ListWorkflowRunsForRepoParameters
+ * @typedef {RestEndpointMethodTypes["actions"]["listWorkflowRunsForRepo"]["response"]["data"]["workflow_runs"][number]} WorkflowRun
+ */
+
+/**
+ *
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ */
+export async function verifyRunStatus({ github, context, core }) {
+  const checkRunName = process.env.CHECK_RUN_NAME || "";
+  if (!checkRunName) {
+    throw new Error("CHECK_RUN_NAME is not set");
+  }
+
+  const workflowName = process.env.WORKFLOW_NAME || "";
+  if (!workflowName) {
+    throw new Error("WORKFLOW_NAME is not set");
+  }
+
+  // Exit early when context is a check_run event and the check run does not
+  // match the checkRunName.
+  if (context.eventName == "check_run") {
+    const contextRunName = context.payload.check_run.name;
+    if (contextRunName !== checkRunName) {
+      const message = `Check run name (${contextRunName}) does not match input: ${checkRunName}`;
+      core.info(message);
+      core.notice(message);
```

Comment:
Why log to both info and notice?

Reply:
Logging to notice produces the output in the workflow summary which makes it easier to skim: 

![image](https://github.com/user-attachments/assets/10829035-15be-4eb5-97e3-2fc756c7504a)

.. But since both log to stdout, I'll remove the `core.info`s